src/mono/netcore/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.Mono.cs

```diff
@@ -1,16 +1,14 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
+using System.Runtime.CompilerServices;
+
 
 namespace System.Globalization
 {
     internal partial class GlobalizationMode
     {
-        internal static bool UseNls => true;
-
-        private static bool GetGlobalizationInvariantMode()
-        {
-            return GetInvariantSwitchValue();
-        }
+        internal static bool Invariant { get; } = GetInvariantSwitchValue();
+        internal static bool UseNls { get; } = !Invariant && Interop.Globalization.LoadICU() == 0;
```

Comment:
Sorry for the partial reviews, but I forgot about this:

Also, customers can disable using ICU and enable using NLS via a config switch. Just the same way as Invariant mode. So before calling Interop.Globalization.LoadICU we should check for the environment variable to see if it is set. DOTNET_SYSTEM_GLOBALIZATION_USENLS

See the coreclr logic: 
https://github.com/dotnet/runtime/blob/master/src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.cs#L13-L15

Reply:
Fixed